### DISCUSSION TITLE
/net API search results inconsistency

### DISCUSSION THREAD

**Discussion by mpounsett** [peeringdb/peeringdb#1444] [Q&A] [answered]
Suppose there's a net object in the database with the following data:
```json
{
   "id": 1337,
   "asn": 64496,
   ...
}
```

Requesting this object from the API, a call to `/api/net?asn=64496` returns a subset of the data returned by `/api/net/1337`.  Specifically, the `org` key and all of the `*_set` keys are missing from the dictionary when using the search rather than direct access.  

It seems like `/net/X` defaults to `/net/X?depth=2`, but `/net?asn=X` defaults to `/net?asn=X&depth=0`.

Is this difference deliberate, or due to an oversight?

**Comment by arnoldnipper** [comment, marked as answer]
On 15.09.2023 01:08, Matthew Pounsett wrote:
> Suppose there's a net object in the database with the following data:
> 
> {
>     "id":1337,
>     "asn":64496,
>     ...
> }
> 
> Requesting this object from the API, a call to |/api/net?asn=64496| 
> returns a subset of the data returned by |/api/net/1337|. Specifically, 
> the |org| key and all of the |*_set| keys are missing from the 
> dictionary when using the search rather than direct access.
> 
> It seems like |/net/X| defaults to |/net/X?depth=2|, but |/net?asn=X| 
> defaults to |/net?asn=X&depth=0|.
> 
> Is this difference deliberate, or due to an oversight?
> 

Afair, this is according to the documentation [0]. Looks a bit weird, though

  * GET: single object ... depth int defaults to 2 aka. nested sets and 
objects will be expande

  * GET: multiple objects ... 0: don't expand anything (default)


Arnold

[0] https://docs.peeringdb.com/api_specs/
-- 
Keep calm, keep distance, keep connected!

Arnold Nipper
email: ***@***.***
mobile: +49 172 2650958

**Reply by mpounsett** [reply]
Oh, that is a different set of API specs from the ones linked off the PeeringDB web site.  That caveat doesn't seem to be in the main API docs at https://www.peeringdb.com/apidocs

Thanks, that clarifies things.  It's arguable that `/api/net?asn=X` should only ever return a single object, but I can see how it's an edge case of other searches, which would be more likely to return multiple objects.

**Reply by arnoldnipper** [reply]
I guess the logic is the other way around and makes perfect sense. Querying for /net may return n objects, 0 <=n <= size(net). Querying for /net/$id returns at most 1 object.
